modules/test/playwright/tests/headless-builder-web/schemas.spec.ts

```diff
@@ -61,221 +154,135 @@ testFeatureFlagsDisabled('can see all available object defitions on schema creat
 					},
 				],
 				pluralLabel: {
-					en_US: `objectDefinitions${i}`,
+					en_US: `objectDefinitions`,
 				},
 				portlet: true,
 				scope: 'company',
 				status: {
 					code: 0,
 				},
-			})
-		);
-	}
+			});
 
-	const application = await apiHelpers.object.postObjectEntry(
-		{
-			apiApplicationToAPISchemas: [
-				{
-					description: 'API Application Schema',
-					externalReferenceCode: 'api-application-schema',
-					mainObjectDefinitionERC: 'L_API_APPLICATION',
-					name: 'API Application Schema',
-				},
-			],
-			applicationStatus: 'published',
-			baseURL: 'basic-application',
-			description: 'Test API Application',
-			externalReferenceCode: 'basic-application',
-			title: 'Basic application',
-		},
-		'headless-builder/applications'
-	);
+		const application = await apiHelpers.object.postObjectEntry(
+			{
+				apiApplicationToAPISchemas: [
+					{
+						description: 'API Application Schema',
+						externalReferenceCode: 'api-application-schema',
+						mainObjectDefinitionERC: 'L_API_APPLICATION',
+						name: 'API Application Schema',
+					},
+				],
+				applicationStatus: 'published',
+				baseURL: 'basic-application',
+				description: 'Test API Application',
+				externalReferenceCode: 'basic-application',
+				title: 'Basic application',
+			},
+			'headless-builder/applications'
+		);
 
-	await headlessBuilderPage.goto();
-	await headlessBuilderPage.goToEditApplication(application.title);
-	await applicationPage.goToSchemasTab();
-	await applicationPage.addSchemaButton.click();
-	await applicationPage.schemaObjectDefinitionSelector.click();
+		await headlessBuilderPage.goto();
+		await headlessBuilderPage.goToEditApplication(application.title);
+		await applicationPage.goToSchemasTab();
+		await applicationPage.addSchemaButton.click();
+		await applicationPage.schemaObjectDefinitionSelector.click();
 
-	objectDefinitions.forEach((objectDefinition) => {
 		expect(
-			applicationPage.page.getByRole('menuitem', {
-				exact: true,
-				name: objectDefinition.name,
-			})
-		).toBeVisible();
-	});
+			await applicationPage.page.getByRole('menu').allTextContents()
+		).toEqual([
+			'APISortAPIFilterAPIEndpointAPIPropertyAPISchemaAPIApplicationObjectDefinition',
```

Comment:
I was investigating on how can we get all the items but not in a huuuuuuge string because I think this could be improve it:

```javascript
		const expectObjectDefintion = ['APISort', 'APIFilter', 'APIEndpoint', 'APIProperty', 'APISchema', 'APIApplication', 'ObjectDefinition'];
		
		let actualObjectDefintion = [];
		
		
		await applicationPage.page.getByRole('menu').allInnerTexts().then((arrStr) => {
			arrStr.forEach((ele) => {
				actualObjectDefintion = ele.split('\n');
			});
		  });
		
		expect(expectObjectDefintion).toEqual(actualObjectDefintion);
```
The idea of this is just to test that the `expectedObjectDefintions` array should be the `actualObjectDefintions` array. I think this is a better approach than comparing a huge string. What do you think?

Reply:
Hi @sergiojimcos, I like the idea of splitting the string to make it more readable. What do you think of something like this?
```
expect(
	(
		await applicationPage.page.getByRole('menu').allInnerTexts()
	)[0].split('\n')
).toEqual([
	'APISort',
	'APIFilter',
	'APIEndpoint',
	'APIProperty',
	'APISchema',
	'APIApplication',
	'ObjectDefinition',
]);

```

Reply:
@jaimelr10 yeahhh much better!! thanks